bitcoin-core-dev-tech/2023-09/wallet-coin-selection.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Discussion on open Coin Selection matters
+categories: ['core-dev-tech']
+tags: ['bitcoin core', 'coinselection']
+date: 2023-09-19
+---
+
+* **Topic: review of https://github.com/bitcoin/bitcoin/pull/27601**
+    * Problem statement: when doing manual RBF (without using bumpfee RPC) we treat previous change output as a receiver and thus create two outputs to the same address
+    * Proposal: combine amount on outputs to the same address
+    * What are valid use-cases for having the same address for change and output?
+        * Consolidation with payment
+            * Alternative: Use sendall with two outputs one with an amount and yours without an amount
+        * Payment and send at least X to yourself
+        * Consolidate with automatic coin selection of at least X 
+            * Alternative: Implement "Send at least" as a separate and explicitly feature
+    * Consider just giving an error if change destination is the same with one of the outputs
+        * Should we do the enforcement only at RPC level?
+            * No, because this is risky 
+    * We don't want to fix manual RBF, because we have bumpfee RPC
+        * Maybe you can also take original TX and give to fundrawtransaction
+* **Topic:  https://github.com/bitcoin/bitcoin/pull/26732**
+    * CreateTransactionInternal should remove UTXOs conflicting with preselected  
+
+* **Topic: Where do we make change decisions? During tx building or coin selection?**
+    * Whether we create change, how big it is and the type
+    * We determine the size of the change twice
+        * One time to feed into coinselection 
+        * Second time when we do the actual tx building 
+    * One approach is to let coinselection pick change type
+    * Conclusion so far: Status quo remains
+
+* **Topic: Bug when BnB produce change https://github.com/bitcoin/bitcoin/issues/28180**
+    * Min_viable_change is different from cost_of_change and it doesn't include change_fee
+            * Zeroing out coins selection params for SFFO is bad
+        * We need to fix change_fee
+    * One solution: Reduce upper bound for BnB when SFFO
+    * Another solution: disable BnB and exact knapsack when user requested SFFO
+
+* SFFO use cases are still unclear
```

Comment:
Is this bullet point supposed to be part of the topic above it (Topic: Bug when BnB produce change)?

Reply:
Made it part of previous topic.